src/sys/unix/waker/pipe.rs

```diff
@@ -40,6 +40,7 @@ impl Waker {
         // The epoll emulation on some illumos systems currently requires
         // the pipe buffer to be completely empty for an edge-triggered
         // wakeup on the pipe read side.
+        // See https://www.illumos.org/issues/16700.
```

Comment:
Thanks for adding this link! The bugs are related but actually distinct. The correct issue here is https://www.illumos.org/issues/13436.